### ISSUE TITLE
DependsOn Enchancments

### ISSUE THREAD

**Issue by jacob7395** [thomhurst/TUnit#1038]
Hey, The `DependsOn` feature in TUnit is incredibly useful—it's something I’ve long wanted from other testing frameworks.

One enhancement that would benefit my tests integration of dependencies across different test classes, allowing tests to depend on methods within other test classes. Looking something like;

```csharp
public class ClassASetupTests
{
    [Test]
    public Task Test() {}
}

public class ClassADbTests
{
    [Test]
    [DependsOn(ClassName: nameof(ClassASetupTests), testName: nameof(Test), parameterTypes: [])]
    public Task Test() {}
}
```

This becomes even more helpful when testing at a higher level like with simulation testing.

Additionally, the ability to make an entire test class dependent on another class would be beneficial in certain scenarios. However, this might be a feature best considered later.

Another use case where this functionality would have been useful for me involves organizing test cases with subclasses:

```csharp
public class ClassATests
{
    [Test]
    [DependsOn(ClassName: nameof(ClassASetupTests), testName: nameof(TestClassInitialization), parameterTypes: [])]
    public Task TestClassInitialization() {}

    public class ClassADbTests
    {
        [Test]
        [DependsOn(ClassName: nameof(ClassASetupTests), testName: nameof(TestClassInitialization), parameterTypes: [])]
        public Task Test() {} 
    }

    public class ClassAEventsTests
    {
        [Test]
        [DependsOn(ClassName: nameof(ClassASetupTests.ClassADbTests), testName: nameof(Test), parameterTypes: [])]
        public Task Test() {} 
    }
}
```

While I don’t often organize tests this way, it can be a convenient method for breaking down test classes. IDEs handle these nicely in the test explorer, grouping them in a readable format.

If this is something you’d like TUnit to support, I’d be happy to dedicate some time over the next couple of weeks to work on an implementation.

Before proceeding, it would be great to hear your thoughts and any advice on implementing this feature, as I haven’t had a chance to dive deeply into the TUnit codebase yet.

**Comment by thomhurst** [comment]
I think I could implement both of these fairly easily if you want so you don't need to dedicate time to it.

I'd need to check, but all the infrastructure is already available for locating and waiting for tests, so I would just need to extend the attributes, and extend the TestFinder class to be able to filter by types too

**Comment by jacob7395** [comment]
Amazing, I'll take a look at the PR. It will help me understand a bit of what is going on.

Thanks 😍

**Comment by thomhurst** [comment]
@jacob7395 Should be available in v0.2.60
